src/test/java/cloud/fogbow/accs/core/processors/SyncProcessorTest.java

```diff
@@ -0,0 +1,301 @@
+import org.powermock.modules.junit4.PowerMockRunnerDelegate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+
+@PowerMockIgnore({"javax.management.*"})
+@PowerMockRunnerDelegate(SpringRunner.class)
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(SyncProcessor.class)
+@SpringBootTest
+public class SyncProcessorTest extends ProcessorsBaseUnitTest {
+
+    private final String FAKE_ORDER_ID = "SINGLE_FAKE_ORDER_ID";
+
+    @Autowired
+    private SyncProcessor syncProcessor;
+
+    @MockBean
+    private DatabaseManager dbManager;
+
+    //test case: test if the method works properly with each order state.
+    @Test
+    public void orderHasFinished() {
+        // exercise/verify
+        Assert.assertTrue(syncProcessor.orderHasFinished(OrderState.CLOSED));
+        Assert.assertTrue(syncProcessor.orderHasFinished(OrderState.FAILED_AFTER_SUCCESSFUL_REQUEST));
+        Assert.assertTrue(syncProcessor.orderHasFinished(OrderState.DEACTIVATED));
+        Assert.assertFalse(syncProcessor.orderHasFinished(OrderState.FULFILLED));
+    }
+
+    //test case: checks if getDuration is doing the right math.
+    @Test
+    public void testGetDuration() {
+        //setup
+        Timestamp now = new Timestamp(System.currentTimeMillis());
+        Timestamp future = new Timestamp(System.currentTimeMillis() + 10000);
+
+        //exercise/verify
+        Assert.assertEquals(future.getTime() - now.getTime(), syncProcessor.getDuration(future, now));
+    }
+
+    //test case: edge test. Check if the method deals with null values properly.
+    @Test
+    public void testGetDurationWithNullIntervals() {
+        //exercise/verify
+        Assert.assertEquals(0, syncProcessor.getDuration(null, null));
+    }
+
+    //test case: Check getDuration with equal intervals.
+    @Test
+    public void testGetDurationWithTheSameTimestamp() {
+        //setup
+        Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+        //exercise/verify
+        Assert.assertEquals(0, syncProcessor.getDuration(timestamp, timestamp));
+    }
+
+    //test case: Check if the method keeps record's duration when the order is fulfilled.
+    @Test
+    public void testSetClosedOrderDurationWithFulfilledOrder() {
+        //setup
+        Record rec = new Record();
+        Timestamp startTime = new Timestamp(System.currentTimeMillis());
+        Timestamp endTime = new Timestamp(System.currentTimeMillis() + 10000);
+        rec.setStartTime(startTime);
+        rec.setEndTime(endTime);
+        AuditableOrderStateChange auditableOrderStateChange = new AuditableOrderStateChange();
+        auditableOrderStateChange.setNewState(OrderState.FULFILLED);
```

Comment:
Try to simplify and unify code by extracting common procedures in reusable methods for other tests in the BaseUnitTests class. 

Check this in the other test methods.